hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkInsertOverwriteCommitActionExecutor.java

```diff
@@ -65,9 +70,21 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute() {
 
   @Override
   protected Partitioner getPartitioner(WorkloadProfile profile) {
-    return table.getStorageLayout().layoutPartitionerClass()
-        .map(c -> getLayoutPartitioner(profile, c))
-        .orElse(new SparkInsertOverwritePartitioner(profile, context, table, config));
+    // Fix HUDI-5857, insert overwrite should generate new file group id
+    if (config.getIndexType().equals(BUCKET)) {
+      switch (config.getBucketIndexEngineType()) {
+        case SIMPLE:
+          return new SparkInsertOverwriteSimpleBucketIndexPartitioner(profile, context, table, config);
+        case CONSISTENT_HASHING:
+          return new SparkInsertOverwriteConsistentBucketIndexPartitioner(profile, context, table, config);
+        default:
+          throw new HoodieNotSupportedException("Unknown bucket index engine type: " + config.getBucketIndexEngineType());
```

Comment:
Can we inline all the different handling for `getBucketInfo` into `SparkInsertOverwritePartitioner` ? Let's make the code cleaner.

Reply:
I move part of them which related to `ConsistentBucketIndex`  to `SparkInsertOverwritePartitioner`.
And I left other part which related to `SimpleBucketIndex` in `SparkBucketIndexInsertOverwritePartitioner`.
Because SimpleBucketIndex and ConsistentBucketIndex are different when creates new `BucketInfo`.